panels/region/cc-input-chooser.c

```diff
@@ -1211,3 +1211,30 @@ cc_input_chooser_get_selected (GtkWidget  *chooser,
 
   return TRUE;
 }
+
+void
+cc_input_chooser_set_is_small_screen (GtkWidget *chooser,
+                                      gboolean   is_small_screen)
+{
+        CcInputChooserPrivate *priv = GET_PRIVATE (chooser);
+
+        if (is_small_screen) {
+                guint width, height;
+
+                gtk_window_get_size (gtk_window_get_transient_for (GTK_WINDOW (chooser)), &width, &height);
+                gtk_widget_set_size_request (GTK_WIDGET (chooser), width * MAIN_WINDOW_WIDTH_RATIO, height);
```

Comment:
No, it's needed. Because of the scrolled window, the dialog would shrink to a minimum size, which is quite small. So, something has to have a size request, and I set the dialog's one (although it could be the scrolled window's size request too).